### PR TITLE
Add inferred documentation links in report

### PR DESCRIPTION
**Pull request by TWiStErRob**
image

### CODE REVIEW

File: quality/src/test/kotlin/net/twisterrob/gradle/quality/tasks/HtmlReportTaskTest.kt [hunk 1]
```diff
@@ -334,3 +404,6 @@ class HtmlReportTaskTest : BaseIntgTest() {
 			""".trimIndent()
 	}
 }
+
+val TestInfo.testName: String
+	get() = testClass.map { it.name }.orElse("") + "." + testMethod.map { it.name }.orElse("")
```
Comment (TWiStErRob) [inline quality/src/test/kotlin/net/twisterrob/gradle/quality/tasks/HtmlReportTaskTest.kt:409]: private or move

File: quality/src/test/resources/net/twisterrob/gradle/quality/tasks/ViolationTestResources/everything/pmd-multiple.xml [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+		<violation beginline="8" endline="8" begincolumn="4" endcolumn="22" rule="AvoidPrintStackTrace" ruleset="Best Practices" class="PrintStack" method="bar" externalInfoUrl="https://pmd.github.io/pmd-6.39.0/pmd_rules_java_bestpractices.html#avoidprintstacktrace" priority="3">
+			Avoid printStackTrace(); use a logger call instead.
+		</violation>
+	</file>
+	<file name="{{rootProject}}\src\main\java\WithoutPackage.java">
+		<violation beginline="1" endline="3" begincolumn="1" endcolumn="1" rule="NoPackage" ruleset="Code Style" class="WithoutPackage" externalInfoUrl="https://pmd.github.io/pmd-6.39.0/pmd_rules_java_codestyle.html#nopackage" priority="3">
+			All classes, interfaces, enums and annotations must belong to a named package
+		</violation>
+	</file>
+	<error filename="{{rootProject}}\src\main\AndroidManifest.xml" msg="PMDException: Error while parsing {{rootProject}}\src\main\AndroidManifest.xml">
```
Comment (TWiStErRob) [inline quality/src/test/resources/net/twisterrob/gradle/quality/tasks/ViolationTestResources/everything/pmd-multiple.xml:13]: why is this not showing up?

File: checkers/pmd/src/testFixtures/kotlin/net/twisterrob/gradle/pmd/test/PmdTestResources.kt [hunk 3]
```diff
@@ -24,32 +24,49 @@ class PmdTestResources(
...
+
+		val message2: Regex
+			get() =
+				when {
+					gradleVersion() < GradleVersion.version("5.6.0") -> {
+						Regex(""".*src.main.java.pmd.PrintStack\.java:8:\tAvoid printStackTrace\(\); use a logger call instead.""")
+					}
+					gradleVersion() < GradleVersion.version("7.0.0") -> {
+						Regex(""".*src.main.java.pmd.PrintStack\.java:8:\tAvoid printStackTrace\(\); use a logger call instead.""")
+					}
```
Comment (TWiStErRob) [inline checkers/pmd/src/testFixtures/kotlin/net/twisterrob/gradle/pmd/test/PmdTestResources.kt:67]: same?

File: quality/build.gradle.kts [hunk 4]
```diff
@@ -29,7 +29,18 @@ dependencies {
 	implementation(projects.compat.agp)
 
 	compileOnly(libs.annotations.jetbrains)
-	compileOnly(libs.android.gradle)
+	compileOnly(libs.android.gradle) {
+		configurations.compileOnly {
+			// +--- com.android.tools.build:gradle:7.2.1
+			//|    +--- com.android.tools:sdk-common:30.2.1
```
[the remark this answers is not in the document]
Comment (TWiStErRob) [reply quality/build.gradle.kts:35]: @sonatype-lift ignore https://community.sonatype.com/t/how-do-i-report-false-positives/9078

[the remark this answers is not in the document]
Comment (mattjohnson) [reply quality/build.gradle.kts:35]: Hi @TWiStErRob, this is certainly an interesting comment! Our tool for software component analysis will do a gradle build to determine which dependencies are used and check our database for vulnerabilities. The less sophisticated part is figuring out where the dependency is specified in the file, which uses some simple pattern matching. I'll file a bug with out team to improve the comment experience.

This is also reporting that the vulnerability is in a transitive dependency and not in `common`. You can see the vulnerability details here: https://ossindex.sonatype.org/component/pkg:maven/com.google.guava/guava@30.1-jre

[the remark this answers is not in the document]
Comment (TWiStErRob) [reply quality/build.gradle.kts:35]: @mattjohnson while you're here, this flag seems like an interesting problem, because I didn't add any dependencies in this PR, only [removed them](https://github.com/TWiStErRob/net.twisterrob.gradle/pull/245/files#diff-12eee5795f6ef85a11ee0bd4c47b88c7d5a5b5821283111940f8969369885c4e). So the only reason this pre-existing issue (I've been depending on libs.android.gradle forever in multiple places which uses common transitively) is flagged is because suddenly lift found a place to flag it on? Or maybe more likely that it's just keying by "location in build" rather than the vulnerability / dependency?

[the remark this answers is not in the document]
Comment (mattjohnson) [reply quality/build.gradle.kts:35]: Correct, we build both the PR and the target branches and diff the findings. Because one finding was attributed to a line in PR and the other wasn't, it likely believed them to be distinct. Great corner case to add to our test suite 😄

File: quality/src/test/resources/net/twisterrob/gradle/quality/tasks/ViolationTestResources/everything/pmd-multiple.xml (hunk 2 above)
Comment (TWiStErRob) [reply quality/src/test/resources/net/twisterrob/gradle/quality/tasks/ViolationTestResources/everything/pmd-multiple.xml:13]: https://github.com/TWiStErRob/net.twisterrob.gradle/issues/248
